File: kotlin/src/MosabegheyeAsan.java
import java.math.BigInteger;
import java.util.ArrayList;
import java.util.Scanner;

public class MosabegheyeAsan {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        int Birthdate = scanner.nextInt();
        int chosenNumber = scanner.nextInt();
        double factorial = 1;
        int result = 0;
        int untilBirthdate = Birthdate;
        for (int i = 0; i < untilBirthdate; i++) {
            factorial *= Birthdate;
            Birthdate--;
        }
//        System.out.println(factorial);
        double k = 0;
        while (factorial>0){
            k = factorial%10;
            if(k == chosenNumber){
                result++;
            }
            factorial/=10;
        }
        System.out.print(result);
    }

}

File: kotlin/src/MosabegheyeAsan.kt
import java.util.Scanner

fun main(){
    val scanner = Scanner(System.`in`)
    val birthDate = scanner.nextInt()
    val chosenNumber = scanner.next().single()
    var result = 0;
    val factor = factorial(birthDate)
    val factorToString = factor.toString()
    for (i in 0 until factorToString.length-2){

        if(chosenNumber == factorToString[i]){
            result++
        }
    }
//    println(factor)
    print(result)
}

fun factorial(p:Int): Long {
    var k : Long = 1L
    for (i in p downTo 1){
        k*=i
    }
    return k
}